pkg/dataplane/streamconsumergroup/claim.go

```diff
@@ -126,31 +126,47 @@ func (c *claim) fetchRecordBatches(stopChannel chan struct{}, fetchInterval time
 		c.logger,
 		c.getShardLocationAttempts,
 		nil,
-		&c.getShardLocationBackoff, func(attempt int) (bool, error) {
+		&c.getShardLocationBackoff,
+		func(attempt int) (bool, error, int) {
 			c.currentShardLocation, err = c.getCurrentShardLocation(c.shardID)
 			if err != nil {
 				if common.EngineErrorIsNonFatal(err) {
-					return true, errors.Wrap(err, "Failed to get shard location due to a network error")
+					return true, errors.Wrap(err, "Failed to get shard location due to a network error"), 0
 				}
+
+				// if the error is fatal and requires external resolution,
+				// we don't want to fail; instead, we will inform the user via a log
+				if common.EngineErrorIsFatal(err) {
+					// although RetryFunc already logs the error, it logs it as a warning
```

Comment:
It actually doesn't log the error?

Reply:
https://github.com/v3io/v3io-go/blob/0d34f118e16114e122d5b01b33b5c575bd435677/pkg/common/helper.go#L71-L81

Reply:
@TomerShor right 🤦 

i'll remove comment